File: frontend/src/types/Stock.ts
// --- Modelo de datos para un Stock ---
export interface Stock {
  id: string           // ID único del stock
  ticker: string       // Símbolo bursátil
  target_from: string  // Precio objetivo inicial
  target_to: string    // Precio objetivo final
  company: string      // Nombre de la compañía
  action: string       // Acción recomendada
  brokerage: string    // Casa de bolsa
  rating_from: string  // Calificación inicial
  rating_to: string    // Calificación final
  time: string         // Fecha y hora de la recomendación
  origin?: string      // (Opcional) Origen del dato
  next_page?: string   // (Opcional) Siguiente página (paginación)
}

File: backend/models/stock/Stock.go
package models

import (
	"time"
)

type APIResponse struct {
	Items    []StockItem `json:"items"`
	NextPage string      `json:"next_page"`
}

type StockItem struct {
	Id         string    `json:"id"`
	Ticker     string    `json:"ticker"`
	TargetFrom string    `json:"target_from"`
	TargetTo   string    `json:"target_to"`
	Company    string    `json:"company"`
	Action     string    `json:"action"`
	Brokerage  string    `json:"brokerage"`
	RatingFrom string    `json:"rating_from"`
	RatingTo   string    `json:"rating_to"`
	Time       time.Time `json:"time"`
	Origin     string    `json:"origin"`
	Next_page  string    `json:"next_page"`
}

type StockItemCreateRequest struct {
	Ticker     string    `json:"ticker"`
	TargetFrom string    `json:"target_from"`
	TargetTo   string    `json:"target_to"`
	Company    string    `json:"company"`
	Action     string    `json:"action"`
	Brokerage  string    `json:"brokerage"`
	RatingFrom string    `json:"rating_from"`
	RatingTo   string    `json:"rating_to"`
	Time       time.Time `json:"time"`
	Origin     string    `json:"origin"`
	NextPage   string    `json:"next_page"`
}
